beacon_node/network/src/nat.rs

```diff
@@ -20,13 +20,14 @@ pub struct UPnPConfig {
     disable_discovery: bool,
 }
 
-impl From<&NetworkConfig> for UPnPConfig {
-    fn from(config: &NetworkConfig) -> Self {
-        UPnPConfig {
-            tcp_port: config.libp2p_port,
-            udp_port: config.discovery_port,
+impl UPnPConfig {
+    pub fn from_config(config: &NetworkConfig) -> Option<Self> {
+        // TODO: need to read more on UPnP over ipv6
```

Comment:
Yeah me too. I think the UPnP library is pretty opaque and we just send it IPs and the router tries to open them for us.

Reply:
for completeness I add it here: the lib we use accepts exclusively ipv4 addresses so this is basically left as work for the future